### PR TITLE
React to breaking changes in StringValues

### PR DESCRIPTION
**Pull request by rynowak**

### CODE REVIEW

File: test/Microsoft.AspNet.Server.KestrelTests/FrameRequestHeadersTests.cs [hunk 1]
```diff
@@ -121,8 +121,8 @@ public void ContainsAndContainsKeyWork()
             IDictionary<string, StringValues> headers = new FrameRequestHeaders();
             var kv1 = new KeyValuePair<string, StringValues>("host", new[] { "localhost" });
             var kv2 = new KeyValuePair<string, StringValues>("custom", new[] { "value" });
-            var kv1b = new KeyValuePair<string, StringValues>("host", new[] { "localhost" });
-            var kv2b = new KeyValuePair<string, StringValues>("custom", new[] { "value" });
+            var kv1b = new KeyValuePair<string, StringValues>("host", new[] { "not-localhost" });
+            var kv2b = new KeyValuePair<string, StringValues>("custom", new[] { "not-value" });
```
Comment (rynowak) [inline test/Microsoft.AspNet.Server.KestrelTests/FrameRequestHeadersTests.cs]: This test was verifying the wrong thing.

File: src/Microsoft.AspNet.Server.Kestrel/Http/FrameHeaders.cs [hunk 2]
```diff
@@ -152,7 +152,7 @@ IEnumerator IEnumerable.GetEnumerator()
             StringValues value;
             return
                 TryGetValueFast(item.Key, out value) &&
-                object.Equals(value, item.Value) &&
+                value.Equals(item.Value) &&
```
Comment (rynowak) [inline src/Microsoft.AspNet.Server.Kestrel/Http/FrameHeaders.cs]: The old check here could never return true. You're boxing two structs to compare them for reference equality. That's always false.

**Comment by Tratcher** [comment]
:shipit:

**Comment by troydai** [comment]
:shipit: 

I'm looking for any other similar wrong use of value type in Kestrel.

**Pull request merged**